src/Interpreters/JoinToSubqueryTransformVisitor.cpp

```diff
@@ -73,16 +74,28 @@ class ExtractAsterisksMatcher
             }
         }
 
-        void addTableColumns(const String & table_name)
+        using ShouldAddColumnPredicate = std::function<bool (const String&)>;
+
+        /// Add columns from table with table_name into select expression list
+        /// Use should_add_column_predicate for check if column name should be added
+        void addTableColumns(
+            const String & table_name,
+            std::function<bool (const String&)> should_add_column_predicate = [](const String &){ return true; } )
```

Comment:
If we use `= {}` by default, then we can just check `if (should_add_column_predicate && should_add_column_predicate(column.name))` later

Reply:
If we does not use {} and will use return true by default, there will be no additional checks.